Remotion/Data/DomainObjects/Persistence/Rdbms/Parameters/SimpleTypeQueryParameterRecordDefinitionFinder.cs

```diff
@@ -0,0 +1,70 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Linq;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model;
+using Remotion.Data.DomainObjects.Queries;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.Persistence.Rdbms.Parameters;
+
+public class SimpleTypeQueryParameterRecordDefinitionFinder : IQueryParameterRecordDefinitionFinder
+{
+  public IRdbmsStructuredTypeDefinitionFinder StructuredTypeDefinitionFinder { get; }
+
+  public SimpleTypeQueryParameterRecordDefinitionFinder (IRdbmsStructuredTypeDefinitionFinder structuredTypeDefinitionFinder)
+  {
+    ArgumentUtility.CheckNotNull(nameof(structuredTypeDefinitionFinder), structuredTypeDefinitionFinder);
+    StructuredTypeDefinitionFinder = structuredTypeDefinitionFinder;
+  }
+
+  public RecordDefinition? GetRecordDefinition (QueryParameter queryParameter, IQuery query)
+  {
+    ArgumentUtility.CheckNotNull(nameof(queryParameter), queryParameter);
+    ArgumentUtility.CheckNotNull(nameof(query), query);
+
+    IRdbmsStructuredTypeDefinition? structuredTypeDefinition;
+    try
+    {
+      structuredTypeDefinition = StructuredTypeDefinitionFinder.GetTypeDefinition(queryParameter, query);
+      if (structuredTypeDefinition == null)
+        return null;
+    }
+    catch (NotSupportedException ex)
+    {
+      throw new InvalidOperationException(
+          $"The parameter value's type '{queryParameter.Value!.GetType()}' "
```

Comment:
Since there is no way to find a `RecordDefinition` for a `null` value (because we don't have a `Type`), we can check for that at the start of the method and just return `null`.

Reply:
Wait, shouldn't the StrctTypeDefFinder return null in that case? or is this specific to the SimpleType?

Reply:
Checked and yes, that's what's happening. Okay, hmm... this is not great. I don't want to add a guard condition before GetTypeDef since it's not this classes responsibility. this means we really should do this in the exception message because we're using data in there that's not used anywhere else in the method.